src/test/test.cpp

```diff
@@ -600,69 +614,88 @@ recordOrCheckGlobalTestTxMetadata(TransactionMeta const& txMeta)
 }
 
 static void
-loadTestTxMeta(std::string const& path)
+loadTestTxMeta(stdfs::path const& dir)
 {
-    std::ifstream in(path);
-    if (!in)
+    if (!stdfs::is_directory(dir))
     {
-        logErrAndThrow(fmt::format("Failed to open {}", path));
+        logErrAndThrow(fmt::format("{} is not a directory", dir));
     }
-    in.exceptions(std::ios::failbit | std::ios::badbit);
-    Json::Value root;
-    in >> root;
     size_t n = 0;
-    for (auto entry = root.begin(); entry != root.end(); ++entry)
+    for (auto const& dirent : stdfs::directory_iterator{dir})
     {
-        std::string name = entry.key().asString();
-        std::vector<Hash> hashes;
-        for (auto const& h : *entry)
+        auto path = dirent.path();
+        std::ifstream in(path);
+        if (!in)
         {
-            ++n;
-            hashes.emplace_back(hexToBin256(h.asString()));
+            logErrAndThrow(fmt::format("Failed to open {}", path));
         }
-        std::reverse(hashes.begin(), hashes.end());
-        auto pair = gTestTxMetadata.emplace(name, hashes);
-        if (!pair.second)
+        in.exceptions(std::ios::failbit | std::ios::badbit);
+        Json::Value root;
+        in >> root;
+        auto basename = path.filename().stem();
+        auto& fileTestCases = gTestTxMetadata[basename];
+        for (auto entry = root.begin(); entry != root.end(); ++entry)
         {
-            logErrAndThrow(
-                fmt::format("Duplicate test TxMeta found for key: {}", name));
+            std::string name = entry.key().asString();
+            std::vector<Hash> hashes;
+            for (auto const& h : *entry)
+            {
+                ++n;
+                hashes.emplace_back(hexToBin256(h.asString()));
+            }
+            std::reverse(hashes.begin(), hashes.end());
+            auto pair = fileTestCases.emplace(name, hashes);
+            if (!pair.second)
+            {
+                logErrAndThrow(
+                    fmt::format("Duplicate test TxMeta found for key: {}:{}",
+                                basename, name));
+            }
         }
     }
     LOG_INFO(DEFAULT_LOG, "Loaded {} TxMetas to check during replay", n);
 }
 
 static void
-saveTestTxMeta(std::string const& path)
+saveTestTxMeta(stdfs::path const& dir)
 {
-    Json::Value root;
-    for (auto const& pair : gTestTxMetadata)
+    for (auto const& filePair : gTestTxMetadata)
     {
-        Json::Value& hashes = root[pair.first];
-        for (auto const& h : pair.second)
+        Json::Value fileRoot;
+        for (auto const& testCasePair : filePair.second)
+        {
+            Json::Value& hashes = fileRoot[testCasePair.first];
+            for (auto const& h : testCasePair.second)
+            {
+                hashes.append(binToHex(h));
+            }
+        }
+        stdfs::path path = dir / filePair.first;
+        path.replace_extension("json");
```

Comment:
nice - almost makes me cry to see a standard way of replacing extensions

Reply:
Yes :)